istioctl/pkg/ztunnelconfig/ztunnelconfig.go

```diff
@@ -0,0 +1,486 @@
+	workloadConfigCmd.PersistentFlags().StringVar(&address, "address", "", "Filter workloads by address field")
+	workloadConfigCmd.PersistentFlags().StringVar(&node, "node", "", "Filter workloads by node field")
+	workloadConfigCmd.PersistentFlags().BoolVar(&verboseProxyConfig, "verbose", true, "Output more information")
+	workloadConfigCmd.PersistentFlags().StringVarP(&configDumpFile, "file", "f", "",
+		"Ztunnel config dump JSON file")
+	workloadConfigCmd.PersistentFlags().StringVar(&workloadsNamespace, "workloads-namespace", "",
+		"Filter workloads by namespace field")
+
+	return workloadConfigCmd
+}
+
+// Level is an enumeration of all supported log levels.
+type Level int
+
+const (
+	defaultLoggerName       = "level"
+	defaultEnvoyOutputLevel = WarningLevel
+)
+
+const (
+	// OffLevel disables logging
+	OffLevel Level = iota
+	// CriticalLevel enables critical level logging
+	CriticalLevel
+	// ErrorLevel enables error level logging
+	ErrorLevel
+	// WarningLevel enables warning level logging
+	WarningLevel
+	// InfoLevel enables info level logging
+	InfoLevel
+	// DebugLevel enables debug level logging
+	DebugLevel
+	// TraceLevel enables trace level logging
+	TraceLevel
+)
+
+var levelToString = map[Level]string{
+	TraceLevel:    "trace",
+	DebugLevel:    "debug",
+	InfoLevel:     "info",
+	WarningLevel:  "warning",
+	ErrorLevel:    "error",
+	CriticalLevel: "critical",
+	OffLevel:      "off",
+}
+
+var stringToLevel = map[string]Level{
+	"trace":    TraceLevel,
+	"debug":    DebugLevel,
+	"info":     InfoLevel,
+	"warning":  WarningLevel,
+	"warn":     WarningLevel,
+	"error":    ErrorLevel,
+	"critical": CriticalLevel,
+	"off":      OffLevel,
+}
+
+var (
+	loggerLevelString = ""
+	reset             = false
+)
+
+func ztunnelLogLevel(level string) string {
+	switch level {
+	case "warning":
+		return "warn"
+	case "critical":
+		return "error"
+	default:
+		return level
+	}
+}
+
+func logCmd(ctx cli.Context) *cobra.Command {
+	var podNamespace string
+	var podNames []string
+
+	logCmd := &cobra.Command{
+		Use:   "log [<type>/]<name>[.<namespace>]",
+		Short: "Retrieves logging levels of the Envoy in the specified pod",
+		Long:  "Retrieve information about logging levels of the Envoy instance in the specified pod, and update optionally",
+		Example: `  # Retrieve information about logging levels for a given pod from Envoy.
+  istioctl proxy-config log <pod-name[.namespace]>
+
+  # Update levels of the all loggers
+  istioctl proxy-config log <pod-name[.namespace]> --level none
+
+  # Update levels of the specified loggers.
+  istioctl proxy-config log <pod-name[.namespace]> --level http:debug,redis:debug
+
+  # Reset levels of all the loggers to default value (warning).
+  istioctl proxy-config log <pod-name[.namespace]> -r
+`,
+		Aliases: []string{"o"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if labelSelector == "" && len(args) < 1 {
```

Comment:
I would drop this, doesn't make sense for ztunnel. Also aligns with other comments to align the selection of ztunnel across commands